docs/source/guide/custom_metric.md

```diff
@@ -105,4 +107,137 @@ After adding your code to Label Studio Enterprise, the following could happen:
 - Your code might fail to deploy if there are errors. If you don't see errors in the Label Studio UI, check your web browser console after you attempt to save the code.
 - Your code might fail to run properly based on the format of your JSON annotations. Export an example annotation from your project in Label Studio JSON format and make sure your function handles the annotation as expected outside of Label Studio. 
 
-If you change the labeling configuration for your project, you might need to update the custom agreement metric code to handle the new format of annotations produced.
\ No newline at end of file
+If you change the labeling configuration for your project, you might need to update the custom agreement metric code to handle the new format of annotations produced.
+
+## Set up permissions for a private cloud custom agreement metric
+
+If you have Label Studio Enterprise deployed in a private cloud (self-managed) Amazon Web Services (AWS) Elastic Compute Cluster (EC2) instance or Amazon Elastic Kubernetes Service (EKS), you must grant additional permissions for the `label-studio` service account to be able to set up and run custom agreement metrics. 
+
+To set up the permissions, do the following: 
+1. Create an AWS IAM role to be used by the custom metric Lambda functions to store logs in Cloudwatch 
+2. Define an IAM policy allowing Label Studio Enterprise to run code in AWS Lambda.
+3. Attach the IAM policy to the `label-studio` service account. How you do this depends on your deployment scenario:
+   - [Deployed with Docker Compose running in EC2](#Deployed-with-Docker-Compose-running-in-EC2)
+   - [Deployed in EKS with an OIDC provider](#Deployed-in-EKS-with-an-OIDC-provider)
+   - [Deployed in EKS without an OIDC provider](#Deployed-in-EKS-without-an-OIDC-provider)
+
+You must know your AWS account ID to perform these steps. 
+
+### Create an AWS IAM role
+
+Using your preferred method, create an AWS IAM role. 
+
+1. Create an AWS IAM role named `LSE_CustomMetricsExecuteRole`. For guidance, see the [AWS Identity and Access Management doc for Creating IAM roles](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_create.html)
```

Comment:
any additional guidance needed here? any specific permissions that the role would need beyond the policy attached in step 2?

Reply:
only policy from step 2, nothing else